lib/crypto/src/crypto.erl

```diff
@@ -47,533 +56,572 @@
+    brainpoolP192t1 |
+    brainpoolP224r1 |
+    brainpoolP224t1 |
+    brainpoolP256r1 |
+    brainpoolP256t1 |
+    brainpoolP320r1 |
+    brainpoolP320t1 |
+    brainpoolP384r1 |
+    brainpoolP384t1 |
+    brainpoolP512r1 |
+    brainpoolP512t1 |
+    c2pnb163v1 |
+    c2pnb163v2 |
+    c2pnb163v3 |
+    c2pnb176v1 |
+    c2pnb208w1 |
+    c2pnb272w1 |
+    c2pnb304w1 |
+    c2pnb368w1 |
+    c2tnb191v1 |
+    c2tnb191v2 |
+    c2tnb191v3 |
+    c2tnb239v1 |
+    c2tnb239v2 |
+    c2tnb239v3 |
+    c2tnb359v1 |
+    c2tnb431r1 |
+    ipsec3 |
+    ipsec4 |
+    prime192v1 |
+    prime192v2 |
+    prime192v3 |
+    prime239v1 |
+    prime239v2 |
+    prime239v3 |
+    prime256v1 |
+    secp112r1 |
+    secp112r2 |
+    secp128r1 |
+    secp128r2 |
+    secp160k1 |
+    secp160r1 |
+    secp160r2 |
+    secp192k1 |
+    secp192r1 |
+    secp224k1 |
+    secp224r1 |
+    secp256k1 |
+    secp256r1 |
+    secp384r1 |
+    secp521r1 |
+    sect113r1 |
+    sect113r2 |
+    sect131r1 |
+    sect131r2 |
+    sect163k1 |
+    sect163r1 |
+    sect163r2 |
+    sect193r1 |
+    sect193r2 |
+    sect233k1 |
+    sect233r1 |
+    sect239k1 |
+    sect283k1 |
+    sect283r1 |
+    sect409k1 |
+    sect409r1 |
+    sect571k1 |
+    sect571r1 |
+    wtls1 |
+    wtls10 |
+    wtls11 |
+    wtls12 |
+    wtls3 |
+    wtls4 |
+    wtls5 |
+    wtls6 |
+    wtls7 |
+    wtls8 |
+    wtls9.
+
+-type edwards_curve_dh() :: x25519 | x448.
+
+-type edwards_curve_ed() :: ed25519 | ed448.
 
 %%%----------------------------------------------------------------
 %%% New cipher schema
 %%%
--type cipher() :: cipher_no_iv()
-                | cipher_iv()
-                | cipher_aead() .
-
--type cipher_no_iv() :: aes_128_ecb
-                      | aes_192_ecb
-                      | aes_256_ecb
-
-                      | blowfish_ecb
-                      | des_ecb
-                      | rc4 .
-
--type cipher_iv() :: aes_128_cbc
-                   | aes_192_cbc
-                   | aes_256_cbc
-
-                   | aes_128_cfb128
-                   | aes_192_cfb128
-                   | aes_256_cfb128
-
-                   | aes_128_cfb8
-                   | aes_192_cfb8
-                   | aes_256_cfb8
-
-                   | aes_128_ctr
-                   | aes_192_ctr
-                   | aes_256_ctr
-
-                   | aes_ige256
-
-                   | blowfish_cbc
-                   | blowfish_cfb64
-                   | blowfish_ofb64
-                   | chacha20
-                   | des_ede3_cbc
-                   | des_ede3_cfb
-
-                   | des_cbc
-                   | des_cfb
-                   | rc2_cbc .
-
-
--type cipher_aead() :: aes_128_ccm
-                     | aes_192_ccm
-                     | aes_256_ccm
-
-                     | aes_128_gcm
-                     | aes_192_gcm
-                     | aes_256_gcm
-
-                     | chacha20_poly1305 .
-
+-type cipher() ::
+    cipher_no_iv() |
```

Comment:
Is "`|` before atom" a choice?

Reply:
I mean, I don't actually think that
```erlang
-type this() :: 'OR' |
                that.
```
(or even)
```erlang
-type this() ::
    'OR' |
    that.
```
is much less readable than
```erlang
-type this() :: 'OR'
              | that.
```
but I guess there are different preferences.

Reply:
Since this is a repeated issue, I created an issue for a centralized discussion https://github.com/WhatsApp/erlfmt/issues/167

Reply:
FYI: Data analysis is suggesting we move pipes to before the "atom"
https://github.com/WhatsApp/erlfmt/pull/186